daemon/mgr/container.go

```diff
@@ -1175,6 +1179,47 @@ func (mgr *ContainerManager) Restart(ctx context.Context, name string, timeout i
 	return mgr.start(ctx, c, "")
 }
 
+// DisconnectContainerFromNetwork disconnects the given container from
+// given network
+func (mgr *ContainerManager) DisconnectContainerFromNetwork(ctx context.Context, containerName, networkName string, force bool) error {
+	c, err := mgr.container(containerName)
+	if err != nil {
+		// TODO(ziren): if force is true, force delete endpoint
+		return err
+	}
+
+	// Get network
+	network, err := mgr.NetworkMgr.Get(ctx, networkName)
+	if err != nil {
+		return fmt.Errorf("failed to disconnect container %s from %s: get network failed: %v", c.Name(), networkName, err)
```

Comment:
`fmt.Errorf("failed to get network %s when disconnecting container %s from it: %v", c.Name(), network, err)` ?